lua/starfall/libs_sh/darkrp.lua

```diff
@@ -0,0 +1,925 @@
+local checkluatype = SF.CheckLuaType
+local checkpattern = SF.CheckPattern
+local registerprivilege = SF.Permissions.registerPrivilege
+
+-- checksafety and assertsafety are here to ensure that even if DarkRP's API changes, it won't create a security vulnerability.
+local whitelist = {
+	["nil"] = true,
+	["boolean"] = true,
+	["number"] = true,
+	["string"] = true,
+}
+local pairs = pairs
+local select = select
+local type = type
+local unpack = unpack
+local function checksafety(...)
+	for k=1, select("#", ...) do
+		local v = select(k, ...)
+		if not whitelist[type(v)] then
+			return false
+		end
+	end
```

Comment:
select is a slow function. This is actually faster.
```lua
for k, v in ipairs{...} do
    if not whitelist[type(v)] then
        return false
    end
end
```

Reply:
Benchmarked it, with JIT off, `pairs` and `ipairs` are both consistently faster than `select` (with `pairs` being fastest obviously). With JIT on, `select` and `pairs` are about tied, although `pairs` and `ipairs` have a tendency to be considerably slower, but only about once every five tests. Probably a JIT quirk (I'm on 64-bit Linux). I'm going to use `pairs`.